packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -90,6 +103,44 @@ async function getEventTypeId({
   return eventType?.id;
 }
 
+async function _getReservedSlotsAndCleanupExpired({
+  bookerClientUid,
+  usersWithCredentials,
+  eventTypeId,
+}: {
+  bookerClientUid: string | undefined;
+  usersWithCredentials: GetAvailabilityUser[];
+  eventTypeId: number;
+}) {
+  const currentTimeInUtc = dayjs.utc().format();
+
+  const unexpiredSelectedSlots =
+    (await prisma.selectedSlots.findMany({
+      where: {
+        userId: { in: usersWithCredentials.map((user) => user.id) },
+        releaseAt: { gt: currentTimeInUtc },
+      },
+      ...selectSelectedSlots,
+    })) || [];
+
+  const slotsSelectedByOtherUsers = unexpiredSelectedSlots.filter((slot) => slot.uid !== bookerClientUid);
+
+  await _cleanupExpiredSlots({ eventTypeId });
+
+  const reservedSlots = slotsSelectedByOtherUsers;
+
+  return reservedSlots;
+
+  async function _cleanupExpiredSlots({ eventTypeId }: { eventTypeId: number }) {
+    await prisma.selectedSlots.deleteMany({
+      where: {
+        eventTypeId: { equals: eventTypeId },
+        releaseAt: { lt: currentTimeInUtc },
```

Comment:
Change: We were earlier cleaning up other users selected slots as well. Now we clean this eventType's expired slots only.

I think we could even remove eventTypeId condition, but deletion could be slow query and don't want to slow down availability request